src/components/Search/FilterDropdowns/UserSelectPopup.tsx

```diff
@@ -0,0 +1,208 @@
+import isEmpty from 'lodash/isEmpty';
+import React, {memo, useCallback, useEffect, useMemo, useState} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import Button from '@components/Button';
+import {usePersonalDetails} from '@components/OnyxProvider';
+import {useOptionsList} from '@components/OptionListContextProvider';
+import SelectionList from '@components/SelectionList';
+import UserSelectionListItem from '@components/SelectionList/Search/UserSelectionListItem';
+import useDebouncedState from '@hooks/useDebouncedState';
+import useLocalize from '@hooks/useLocalize';
+import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import type {Option, Section} from '@libs/OptionsListUtils';
+import {filterAndOrderOptions, getValidOptions} from '@libs/OptionsListUtils';
+import type {OptionData} from '@libs/ReportUtils';
+import {searchInServer} from '@userActions/Report';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+function getSelectedOptionData(option: Option) {
+    return {...option, selected: true};
+}
+
+type UserSelectPopupProps = {
+    /** The currently selected users */
+    value: string[];
+
+    /** Function to call to close the overlay when changes are applied */
+    closeOverlay: () => void;
+
+    /** Function to call when changes are applied */
+    onChange: (value: string[]) => void;
+};
+
+function UserSelectPopup({value, closeOverlay, onChange}: UserSelectPopupProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const {options} = useOptionsList();
+    const personalDetails = usePersonalDetails();
+    const {windowHeight} = useWindowDimensions();
+    const {shouldUseNarrowLayout} = useResponsiveLayout();
+    const [session] = useOnyx(ONYXKEYS.SESSION, {canBeMissing: true});
+
+    // Since accountIDs are passed as value, we need to "populate" them into OptionData
+    const initialSelectedData: OptionData[] = useMemo(() => {
+        const initialOptions = value
+            .map((accountID) => {
+                const participant = personalDetails?.[accountID];
+
+                if (!participant) {
+                    return;
+                }
+
+                return getSelectedOptionData(participant);
+            })
+            .filter(Boolean) as OptionData[];
```

Comment:
Can you avoid the cast `as OptionData[]` and instead use a `reduce`? Casting defeats the purpose of having a typed language.